sqlserver/datadog_checks/sqlserver/metrics.py

```diff
@@ -583,14 +583,11 @@ def fetch_metric(self, rows, columns):
         value_column_index = columns.index(self.column)
 
         for row in rows:
-            if row[database_name] != self.instance:
```

Comment:
ah wow ... any idea why this logic was in place? I'm assuming this is causing _a lot_ of the default int sqlserver metrics to mis-tagged

Reply:
> any idea why this logic was in place? 

Good question. A speculation: the code is being ran once for each monitored database and then the `state` and `recovery_model` only for this database are emitted. I'd need to verify that. 

I see now a problem with my solution: after removing `if/continue` the metric will be emitted also for the databases that aren't being monitored.

I have to get a better understanding of the existing logic, i.e. first figure out how `self.instance` in `SqlDatabaseStats` is initialised .

Reply:
yeah for sure, I am wondering what else this breaks. If you see an issue with the logic, but our tests are passing this is also very bad lol 

Need better test coverage for this integration for sure

Reply:
> yeah for sure, I am wondering what else this breaks. 
I think the only problem is that `state` and `recovery_model` are emitted for the databases that are not monitored, and also that we will emit the same metric multiple times, which might become a problem on instances with many databases.

I'll go over it once again and also write missing test cases.

Reply:
I've revised my findings in the PR description.